src/include/catalog/catalog.h

```diff
@@ -102,9 +102,11 @@ class CatalogContent {
     const Property& getRelProperty(
         common::table_id_t tableID, const std::string& propertyName) const;
 
-    std::vector<Property> getAllNodeProperties(common::table_id_t tableID) const;
+    inline const std::vector<Property>& getNodeProperties(common::table_id_t tableID) const {
```

Comment:
I'm thinking if we should enforce a naming style for returning reference of a value, like `getNodePropertiesRef`, etc. What do you think?